sqlglot/optimizer/expand_multi_table_selects.py

```diff
@@ -1,26 +1,63 @@
 import sqlglot.expressions as exp
+from sqlglot.helper import tsort
 
 
 def expand_multi_table_selects(expression):
     for from_ in expression.find_all(exp.From):
         parent = from_.parent
         where = parent.args.get("where")
-        for query in from_.args["expressions"][1:]:
+        tail = from_.args["expressions"][1:]
+
+        for query in tail:
             alias = query.alias_or_name
 
-            predicates = [
-                predicate
-                for predicate in [
-                    column.find_ancestor(exp.PREDICATES)
-                    for column in (where.find_all(exp.Column) if where else [])
-                    if column.text("table") == alias
-                ]
-                if predicate
-            ]
+            predicates = []
+
+            if where:
+                condition = where.this.unnest()
+
+                if isinstance(condition, exp.Connector):
+                    condition = condition.flatten()
+                else:
+                    condition = [condition]
+
+                for predicate in condition:
+                    for column in predicate.find_all(exp.Column):
+                        if column.text("table") == alias:
+                            predicates.append(predicate)
 
             for predicate in predicates:
                 predicate.replace(exp.TRUE)
-            parent.join(query, on=predicates, copy=False)
+            parent.join(
+                query,
+                on=predicates,
+                join_type=None if predicates else "CROSS",
+                copy=False,
+            )
             from_.args["expressions"].remove(query)
 
+        if tail:
+            reorder_joins(from_)
+
     return expression
+
+
+def reorder_joins(from_):
```

Comment:
I think a small comment on what this does would be helpful, since "reorder" doesn't really explain the resulting order.